api_v3/lib/types/filters/ValidateAccessResponseProfile.php

```diff
@@ -88,4 +90,16 @@ private static function validate($clazz)
 		return true;
 	}
 
+	/**
+	 * @return bool
+	 * @throws Exception
+	 */
+	private static function PartnerExcludedFromResponseProfile()
+	{
+		$partnerId = kCurrentContext::$partner_id ? kCurrentContext::$partner_id : kCurrentContext::$ks_partner_id;
```

Comment:
Use kCurrentContext::getCurrentPartnerId it does exaclt what you are doing here.